src/main/java/com/entropyteam/entropay/employees/services/EmployeeService.java

```diff
@@ -84,8 +84,10 @@ protected EmployeeDto toDTO(Employee entity) {
         Integer availableDays = vacationRepository.getAvailableDays(entity.getId());
         Optional<Contract> activeContract =
                 contractRepository.findContractByEmployeeIdAndActiveIsTrueAndDeletedIsFalse(entity.getId());
+        Optional<Contract> latestContract =
+                contracts.stream().max(Comparator.comparing(Contract::getStartDate));
```

Comment:
If you have the active contract you don't need to get the lastest. Also active contract = lastest contract, so:
```
Optional<Contract> latestContract =  contractRepository.findContractByEmployeeIdAndActiveIsTrueAndDeletedIsFalse(entity.getId());
if (latestContract.isEmpty()) {
  latestContract = contracts.stream().max(Comparator.comparing(Contract::getStartDate));
}
```